src/main/kotlin/io/perfometer/runner/ThreadPoolScenarioRunner.kt

```diff
@@ -1,62 +1,95 @@
 package io.perfometer.runner
 
 import io.perfometer.dsl.HttpStep
+import io.perfometer.dsl.ParallelStep
 import io.perfometer.dsl.PauseStep
 import io.perfometer.dsl.RequestStep
 import io.perfometer.http.client.HttpClient
+import io.perfometer.internal.helper.decorateInterruptable
+import io.perfometer.internal.helper.decorateSuspendingInterruptable
 import io.perfometer.statistics.PauseStatistics
 import io.perfometer.statistics.ScenarioSummary
 import kotlinx.coroutines.runBlocking
 import java.time.Duration
-import java.util.concurrent.CompletableFuture
-import java.util.concurrent.Executors
-import java.util.concurrent.TimeUnit
+import java.util.concurrent.*
 
 internal class ThreadPoolScenarioRunner(
     httpClient: HttpClient,
 ) : BaseScenarioRunner(httpClient) {
 
+    // todo @ttarczynski - consider moving this to a separate class
+    private val parallelJobs = ConcurrentLinkedDeque<CompletableFuture<Void>>()
```

Comment:
Queue of parallel jobs should be ThreadLocal (like HttpClient -- see in cookies support implementation)